File: actojat-transpiler/src/main/java/de/netherspace/apps/actojat/intermediaterepresentation/java/ForLoop.java
package de.netherspace.apps.actojat.intermediaterepresentation.java;

import lombok.Getter;

@Getter
public class ForLoop extends Loop {

  private Assignment loopVariable;
  private String loopCondition;
  private String loopIncrement;

  /**
   * Constructor.
   *
   * @param body          the loop body
   * @param loopVariable  the loop variable
   * @param loopCondition the loop condition
   * @param loopIncrement the loop increment statement
   */
  public ForLoop(Statement[] body, Assignment loopVariable,
                 String loopCondition, String loopIncrement) {
    super(body);
    this.loopVariable = loopVariable;
    this.loopCondition = loopCondition;
    this.loopIncrement = loopIncrement;
  }

}

File: actojat-transpiler/src/main/kotlin/de/netherspace/apps/actojat/ir/java/ForLoop.kt
package de.netherspace.apps.actojat.ir.java

class ForLoop(
        val loopVariable: Assignment,
        val loopCondition: Expression.Condition,
        val loopIncrement: Assignment, // TODO: increment and decrement operators should be supported as well!
        body: Sequence<Statement>,
        comment: String?
) : Loop(body = body, comment = comment)
